server/src/capture.rs

```diff
@@ -0,0 +1,198 @@
+use log::{error, info};
+use serde::Deserialize;
+use simplelog::*;
+use simplelog::{ColorChoice, TermLogger, TerminalMode};
+use std::fs;
+use std::fs::File;
+use std::path::Path;
+use tokio::io::{AsyncBufReadExt, BufReader};
+use tokio::net::TcpListener;
+use tokio::sync::mpsc;
+use tokio_postgres::{Client, NoTls}; // Import necessary types
+
+#[derive(Debug, Deserialize)]
+struct DatabaseConfig {
+    db_name: String,
+    db_ip: String,
+    db_user: String,
+    db_password: String,
+}
+
+#[derive(Deserialize, Debug)]
+struct Event {
```

Comment:
Do you have any docs/info on the postgres db you set up? I assume the fields here much exactly match field names in the DB? Likewise, types must match?